src/components/customers/AddCustomerDrawer.tsx

```diff
@@ -67,14 +82,73 @@ export const AddCustomerDrawer = forwardRef<DrawerRef, AddCustomerDrawerProps>(
           syncWithProvider: customer?.providerCustomer?.syncWithProvider ?? false,
         },
         paymentProvider: customer?.paymentProvider ?? undefined,
+        metadata: customer?.metadata ?? undefined,
       },
       validationSchema: object().shape({
         name: string().required(''),
         externalId: string().required(''),
+        metadata: array().of(
+          object().shape({
+            key: string().test({
+              test: function (value, { createError, path }) {
+                if (!value) {
+                  return false
+                }
+
+                // Check key unicity
+                // Compare key index and last index values
+                if (
+                  arguments[1].from[1]?.value?.metadata
+                    ?.map((m: { key: string }) => m.key)
+                    ?.indexOf(value) !==
+                  arguments[1].from[1]?.value?.metadata
+                    ?.map((m: { key: string }) => m.key)
+                    ?.lastIndexOf(value)
+                ) {
+                  return createError({
+                    path,
+                    message: MetadataErrorsEnum.uniqueness,
+                  })
+                }
+
+                if (value.length > 20) {
```

Comment:
Use a constant maybe for 20, that way you can give it a understandable name

Reply:
Fixed